internal/transport/grpchttp2/http2bridge_test.go

```diff
@@ -0,0 +1,609 @@
+		}
+	}
+}
+
+// Tests and verifies that a Data Frame is correctly written.
+func (s) TestBridge_WriteData(t *testing.T) {
+	c := &testConn{}
+	wantData := "test data"
+	testBuf := [][]byte{[]byte("test"), []byte(" data")}
+	f := NewFramerBridge(c, c, 0, nil)
+	f.WriteData(1, false, testBuf...)
+	wantHdr := FrameHeader{
+		Size:     uint32(len(wantData)),
+		Type:     FrameTypeData,
+		Flags:    0,
+		StreamID: 1,
+	}
+	if errs := checkWrittenHeader(c.wbuf[0:9], wantHdr); len(errs) > 0 {
+		t.Errorf("WriteData():\n%s", errors.Join(errs...))
+	}
+	if string(c.wbuf[9:]) != wantData {
+		t.Errorf("WriteData(): Data: got %q, want %q", string(c.wbuf[9:]), wantData)
+	}
+}
+
+// Tests and verifies that a Headers Frame and all its flag permutations are
+// correctly written.
+func (s) TestBridge_WriteHeaders(t *testing.T) {
+	tests := []struct {
+		name       string
+		endStream  bool
+		endHeaders bool
+	}{
+		{name: "no flags", endStream: false, endHeaders: false},
+		{name: "endheaders", endStream: false, endHeaders: true},
+		{name: "endstream and endheaders", endStream: true, endHeaders: true},
+		{name: "endstream", endStream: true, endHeaders: false},
+	}
+	wantData := "test data"
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf(test.name, test.endStream, test.endHeaders), func(t *testing.T) {
+			c := &testConn{}
+			f := NewFramerBridge(c, c, 0, nil)
+
+			f.WriteHeaders(1, test.endStream, test.endHeaders, []byte(wantData))
+
+			var wantFlags Flag
+			if test.endStream {
+				wantFlags |= FlagHeadersEndStream
+			}
+			if test.endHeaders {
+				wantFlags |= FlagHeadersEndHeaders
+			}
+			wantHdr := FrameHeader{
+				Size:     uint32(len(wantData)),
+				Type:     FrameTypeHeaders,
+				Flags:    wantFlags,
+				StreamID: 1,
+			}
+			if errs := checkWrittenHeader(c.wbuf[0:9], wantHdr); len(errs) > 0 {
+				t.Errorf("WriteHeaders():\n%s", errors.Join(errs...))
+			}
+			if data := string(c.wbuf[9:]); data != wantData {
+				t.Errorf("WriteHeaders(): Data: got %q, want %q", data, wantData)
+			}
+			c.wbuf = c.wbuf[:0]
+		})
+
+	}
+}
+
+// Tests and verifies that a RSTStream Frame is correctly written.
+func (s) TestBridge_WriteRSTStream(t *testing.T) {
+	c := &testConn{}
+	f := NewFramerBridge(c, c, 0, nil)
+	f.WriteRSTStream(1, ErrCodeProtocol)
+
+	wantHdr := FrameHeader{
+		Size:     4,
+		Type:     FrameTypeRSTStream,
+		Flags:    0,
+		StreamID: 1,
+	}
+	if errs := checkWrittenHeader(c.wbuf[0:9], wantHdr); len(errs) > 0 {
+		t.Errorf("WriteRSTStream():\n%s", errors.Join(errs...))
+	}
+	if errCode := readUint32(c.wbuf[9:13]); errCode != uint32(ErrCodeProtocol) {
+		t.Errorf("WriteRSTStream(): SettingID: got %d, want %d", errCode, ErrCodeProtocol)
+	}
+}
+
+// Tests and verifies that a Settings Frame is correctly written.
+func (s) TestBridge_WriteSettings(t *testing.T) {
+	c := &testConn{}
+	f := NewFramerBridge(c, c, 0, nil)
+	f.WriteSettings(Setting{ID: SettingsHeaderTableSize, Value: 200})
```

Comment:
I don't think we really need a check for that, since both the names and the values for these items come from the [HTTP/2 Spec](https://httpwg.org/specs/rfc7540.html#SettingValues). Any change on them would imply breaking the procotol.